### PR TITLE
[flutter_tools] make precache force blow away stamp files

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

update flutter precache --force to delete all stamp files. In the event that a user is hitting a cache issue, this should be easier than re-downloading all artifacts or manually blowing away the cache.

This is probably how it should have worked in the first place

### CODE REVIEW

**Comment by jonahwilliams** [comment]
@christopherfujino for thoughts on whether this conflicts with the existing cache work

**Comment by christopherfujino** [comment]
@jonahwilliams great idea, i was thinking about a new command for this functionality, but this makes sense.

File: packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart [hunk 1]
```diff
@@ -38,159 +38,206 @@ void main() {
   });
 
   testUsingContext('precache should acquire lock', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:40]: can this be `testWithoutContext()`?

Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:40]: All of the createTestCommandRunner tests hit the context due to FlutterCommand. I started refactoring things out, but I got stuck on output preferences and rolled it back a while ago

**Review by christopherfujino — APPROVED** [review]
LGTM other than question about using `testWithoutContext()`

**Pull request merged**